packages/flutter/lib/src/material/divider.dart

```diff
@@ -87,23 +92,26 @@ class Divider extends StatelessWidget {
   /// ```
   static BorderSide createBorderSide(BuildContext context, { Color color, double width = 0.0 }) {
     assert(width != null);
-    return new BorderSide(
+    return BorderSide(
       color: color ?? Theme.of(context).dividerColor,
       width: width,
     );
   }
 
   @override
   Widget build(BuildContext context) {
-    return new SizedBox(
-      height: height,
-      child: new Center(
-        child: new Container(
-          height: 0.0,
-          margin: new EdgeInsetsDirectional.only(start: indent),
-          decoration: new BoxDecoration(
-            border: new Border(
-              bottom: createBorderSide(context, color: color),
+    return RotatedBox(
```

Comment:
rather than a rotation transform, which can be expensive, it's probably simplest just to have the vertical divider done using a "left" border instead of a "bottom" border, and then use the "height" as the width instead of the height.